var/spack/repos/builtin/packages/scotch/package.py

```diff
@@ -4,85 +4,90 @@
 class Scotch(Package):
     """Scotch is a software package for graph and mesh/hypergraph
        partitioning, graph clustering, and sparse matrix ordering."""
+
     homepage = "http://www.labri.fr/perso/pelegrin/scotch/"
-    url      = "http://gforge.inria.fr/frs/download.php/file/34099/scotch_6.0.3.tar.gz"
+    url = "http://gforge.inria.fr/frs/download.php/latestfile/298/scotch_6.0.3.tar.gz"
     list_url = "http://gforge.inria.fr/frs/?group_id=248"
 
     version('6.0.3', '10b0cc0f184de2de99859eafaca83cfc')
+    version('5.1.10b', '9b8622b39c141ecaca4a46298486fd99')
 
     variant('mpi', default=False, description='Activate the compilation of PT-Scotch')
     variant('compression', default=True, description='Activate the posibility to use compressed files')
     variant('esmumps', default=False, description='Activate the compilation of the lib esmumps needed by mumps')
     variant('shared', default=True, description='Build shared libraries')
 
-    depends_on('mpi', when='+mpi')
-    depends_on('zlib', when='+compression')
     depends_on('flex')
     depends_on('bison')
+    depends_on('mpi', when='+mpi')
+    depends_on('zlib', when='+compression')
 
-    def compiler_specifics(self, makefile_inc, defines):
-        if self.compiler.name == 'gcc':
-            defines.append('-Drestrict=__restrict')
-        elif self.compiler.name == 'intel':
-            defines.append('-restrict')
+    def validate(self, spec):
```

Comment:
@nrichart: After looking back at the URLs for these package versions, I've found that you're absolutely correct.  I remember the URLs being much more complicated when I was writing this code originally, but I may have been sampling the wrong URLs or doing something else unusual.  In any event, I think it would be good to use the `url_for_version` that you've defined here in the package to eliminate the existing exception for this class of installations.

One quick question before I proceed: do you know whether or not the `+esmumps` variant of each version has a superset of the functionality in its `~esmumps` counterpart?  If so, it would probably just be better to implement `url_for_version` so that it retrieves the `_esmumps` archive for all older versions to avoid issues with differing checksums between the `+esmumps` and `~esmumps` archives.